rts/motoko-rts/src/float.rs

```diff
@@ -6,33 +6,34 @@ use motoko_rts_macros::ic_mem_fn;
 
 // The meaning of the `mode` parameter is documented in motoko-base, function Float.format()
 #[ic_mem_fn]
-unsafe fn float_fmt<M: Memory>(mem: &mut M, a: f64, prec: u32, mode: u32) -> Value {
-    // prec and mode are tagged small words (`Nat8`s), so we shift 24 bits. See
+unsafe fn float_fmt<M: Memory>(mem: &mut M, a: f64, prec: usize, mode: usize) -> Value {
+    // prec and mode are tagged small words (`Nat8`s), so we shift 56 bits. See
     // `TaggedSmallWord.bits_of_type` in compile.ml.
-    let mode = mode >> 24;
-    let prec = core::cmp::min(prec >> 24, 100) as usize;
+    let mode = mode >> 56;
+    let prec = core::cmp::min(prec >> 56, 100) as usize;
 
     // 320 bytes needed for max precision (1.7e308)
-    let buf = [0u8; 320];
-
-    // NB. Using snprintf because I think only 0 and 3 are supposed by Rust's built-in formatter
-    let fmt = match mode {
-        0 => "%.*f\0",
-        1 => "%.*e\0",
-        2 => "%.*g\0",
-        3 => "%.*a\0",
+    const BUFFER_LENGTH: usize = 320;
+    let buffer = match mode {
+        0 => format!(BUFFER_LENGTH, "{:.*}", prec, a),
+        1 => format!(BUFFER_LENGTH, "{:.*e}", prec, a),
+        2 => format!(BUFFER_LENGTH, "{:.*}", prec, a),
+        3 => panic!("float_fmt: unsupported mode"), // TODO: Support this mode (or deprecate in base library).
+        4 => format!(BUFFER_LENGTH, "{}", a),
         _ => panic!("float_fmt: unrecognized mode"),
     };
 
-    let n_written = libc::snprintf(
```

Comment:
Is snprintf not longer available on Rust64? Or we can't use it because of 32 bit libc?

Reply:
Unfortunately, it is because `libc` is not yet ported to 64-bit. And the majority of the functionality cannot be compiled for `wasm64`.

E.g. when including `snprintf`, the compiler fails:
```
include/wasi/api.h:27:2: error: <wasi/api.h> only supports wasm32; doesn't yet support wasm64
#error <wasi/api.h> only supports wasm32; doesn't yet support wasm64
```